collector/current_op.go

```diff
@@ -0,0 +1,48 @@
+package collector
+
+import (
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	instanceFsyncLockWorker = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "instance",
+		Name:      "fsync_lock_worker",
+		Help:      "The value of the fsync field corresponds to whether the fsyncLockWorker is active or not.",
+	})
+)
+
+// CurrentOp keeps the data returned by the currentOp() method.
+type CurrentOp struct {
+	FsyncLockWorker bool   `bson:"fsyncLock"`
+}
+
+// Export exports the current operation status to be consumed by prometheus.
+func (status *CurrentOp) Export(ch chan<- prometheus.Metric) {
+	var floatVar = float64(0)
+	if (status.FsyncLockWorker){
+		floatVar = float64(1)
+	}
+	instanceFsyncLockWorker.Set(floatVar)
+	instanceFsyncLockWorker.Collect(ch)
+}
+
+// Describe describes the current operation status for prometheus.
+func (status *CurrentOp) Describe(ch chan<- *prometheus.Desc) {
+	instanceFsyncLockWorker.Describe(ch)
+}
+
+// GetCurrentOp returns the current operation info.
+func GetCurrentOp(session *mgo.Session, maxTimeMS int64) *CurrentOp {
+	result := &CurrentOp{}
+	err := session.DB("admin").Run(bson.D{{"currentOp", 1}, {"inprog", 0}, {"info", 0}, {"ok", 0}}, result)
```

Comment:
I'm not sure if I understand it right, but the parameters doesn't look right. Could you document what 
{"inprog", 0}, {"info", 0}, {"ok", 0}
means here? 
per my understanding, the parameter is a filter for the list in 'inprog',  so it means find all the items in the inprog list that has infrog==0 && info==0 && ok==0.  I assume what you want is actually 'I don't want member 'inprog', 'info' and 'ok'.  It does have the effect of making 'inprog' an empty list :-). I think having a single {"notexist",0} will have the same effect.

Reply:
understood, changing to err := session.DB("admin").Run(bson.D{{"currentOp", 1}, {"notexist", 0}}, result)